src/Microsoft.AspNet.Mvc.Core/ModelBinding/ArrayModelBinder.cs

```diff
@@ -25,15 +26,16 @@ public override Task<ModelBindingResult> BindModelAsync([NotNull] ModelBindingCo
             return base.BindModelAsync(bindingContext);
         }
 
-        protected override object CreateEmptyCollection()
+        /// <inheritdoc />
+        public override object CreateEmptyCollection(Type targetType)
         {
             return new TElement[0];
         }
 
         /// <inheritdoc />
-        protected override object GetModel(IEnumerable<TElement> newCollection)
+        protected override object GetModel(Type targetType, IEnumerable<TElement> collection)
         {
-            return newCollection?.ToArray();
+            return collection?.ToArray();
```

Comment:
It'll _never_ be an array since the base `CollectionModelBinder` uses a `List<TElement>` until it needs to copy everything into the final model. The value should not be `null` but this was written defensively originally...